ExternalInterface.call cannot call methods of non-object types.

### Describe the bug

I have [a movie](https://mrcheeze.github.io/homestuck-ruffle-tester/#captchas,880,400) that contains the code:

```actionscript
pageURL = ExternalInterface.call("window.location.href.toString");
var tempvars:Array = pageURL.split("&");
```` 

In flash player, this finds the method by running the javascript `window.location.href.toString` which works. In Ruffle [this actually runs](https://github.com/ruffle-rs/ruffle/blob/1b71e288fd984ace46f69c1a2e27e2e2ed31333b/web/src/lib.rs#L1137) the javascript `Reflect.get(window.location.href,'toString')` which throws an error.

These are not equivalent, because js Reflect.get will fail if you pass *anything* other than an object as the first argument. In this case it fails because href is a string (even though it's a string that has the property in question).

### Expected behavior

[This flash](https://mrcheeze.github.io/homestuck-ruffle-tester/#captchas,880,400) should not print
`ERROR core/src/display_object/movie_clip.rs:2344 Error occured when running AVM2 frame script: RustError("Cannot access property http://adobe.com/AS3/2006/builtin::split of null or undefined")`
in the logs.

### Affected platform

Self-hosted version

### Operating system

Windows 11

### Browser

Mozilla Firefox

### Additional information

_No response_

--- a/web/packages/core/src/ruffle-imports.ts
+++ b/web/packages/core/src/ruffle-imports.ts
@@ -59,3 +59,14 @@ export function copyToAudioBufferInterleaved(
         i += 2;
     }
 }
+
+/**
+ * Gets a property of an arbitrary JavaScript value.
+ * This is necessary because Reflect.get does not work for primitive targets.
+ *
+ * @internal
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getProperty(target: any, key: string): any {
+    return target[key];
+}

--- a/web/src/lib.rs
+++ b/web/src/lib.rs
@@ -1174,6 +1174,12 @@ impl ExternalInterfaceMethod for JavascriptMethod {
     }
 }
 
+#[wasm_bindgen(raw_module = "./ruffle-imports")]
+extern "C" {
+    #[wasm_bindgen(catch, js_name = "getProperty")]
+    pub fn get_property(target: &JsValue, key: &JsValue) -> Result<JsValue, JsValue>;
+}
+
 impl JavascriptInterface {
     fn new(js_player: JavascriptPlayer) -> Self {
         Self { js_player }
@@ -1184,7 +1190,7 @@ impl JavascriptInterface {
         let mut value = root;
         for key in name.split('.') {
             parent = value;
-            value = js_sys::Reflect::get(&parent, &JsValue::from_str(key)).ok()?;
+            value = get_property(&parent, &JsValue::from_str(key)).ok()?;
         }
         if value.is_function() {
             Some(JavascriptMethod {

